From aa4edbd22ac3157535f14b5399218153ff05b502 Mon Sep 17 00:00:00 2001
From: Jianfeng Wang <jianfeng.wang@amlogic.com>
Date: Thu, 23 Apr 2015 10:08:24 +0800
Subject: [PATCH 5560/5965] remote: fix array out of bounds issue

This issue is reported by gcc 4.8 Warning aggressive-loop-optimizations

Below loop has the array out of bounds risk/error. Because it used first
level array index as the loop index limitation. But i is used as second
level array index.

remote_main.c:680:33: error: iteration 6u invokes undefined behavior
   [-Werror=aggressive-loop-optimizations]
   mouse_map[remote->map_num][i] = 0xffff;
                                 ^
remote_main.c:679:2: note: containing loop
  for (i = 0; i < ARRAY_SIZE(mouse_map); i++) {
  ^

Change-Id: If515caf93dad52dc486ac256e577219fcb297225
---
 drivers/amlogic/input/remote/am_remote.c | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/drivers/amlogic/input/remote/am_remote.c b/drivers/amlogic/input/remote/am_remote.c
index 79bc58743f4e..1a5bc5bf3c68 100755
--- a/drivers/amlogic/input/remote/am_remote.c
+++ b/drivers/amlogic/input/remote/am_remote.c
@@ -950,13 +950,11 @@ static int remote_probe(struct platform_device *pdev)
 	remote->time_window[6] = 0x1;
 	remote->time_window[7] = 0x1;
 
-	/* Disable the interrupt for the MPUIO keyboard */
-	for (i = 0; i < ARRAY_SIZE(key_map); i++) {
-		key_map[i] = KEY_RESERVED;
-	}
-	for (i = 0; i < ARRAY_SIZE(mouse_map); i++) {
-		mouse_map[i] = 0xffff;
-	}
+	/* Disable the interrupt for the MPUIO keyboard
+	init the default key map table ,and mouse map table.
+	note KEY_RESERVED==0*/
+	memset(key_map, 0x0, sizeof(key_map));
+	memset(mouse_map, 0xff, sizeof(mouse_map));
 	remote->repeat_delay = 250;
 	remote->repeat_peroid = 33;
 
-- 
2.19.0

